services/placement/service/placementservice.go

```diff
@@ -91,6 +84,33 @@ func (ps placementService) BuildInitialPlacement(
 	return p, ps.ss.SetIfNotExist(ps.service, p)
 }
 
+func (ps placementService) initPlacement(
+	instances []services.PlacementInstance,
+	ids []uint32,
+	rf int,
+) (services.ServicePlacement, error) {
+	if (rf == 0) == ps.opts.Sharded() {
```

Comment:
This is a bit hard to read, can we spell out the two cases to make it a bit more readable?